lib/puppet/provider/iis_application_pool/webadministration.rb

```diff
@@ -24,6 +24,7 @@ def create
     result   = self.class.run(inst_cmd)
     Puppet.err "Error creating apppool: #{result[:errormessage]}" unless result[:exitcode] == 0
     Puppet.err "Error creating apppool: #{result[:errormessage]}" unless result[:errormessage].nil?
+    @resource.provider = self.class.instances.find{|prov| prov.name == @resource[:name]}
```

Comment:
I didn't find any way to re-fetch the pool and be anywhere near as fast, though I did find #45 to cut the current time to create / destroy in half. Results:
```
time to create 305 with modified exists? Notice: Applied catalog in 247.87 seconds
time to destroy 305 with standard code Notice: Applied catalog in 577.50 seconds
time to destroy 305 with modified exists? Notice: Applied catalog in 257.48 seconds
time to create 305 with modified create Notice: Applied catalog in 1283.89 seconds
time to create 305 with batched create and ps filtering Notice: Applied catalog in 1382.19 seconds
```